### DISCUSSION TITLE
Stb_image loading animated gifs

### DISCUSSION THREAD

**Discussion by Platforming-Mayhem** [nothings/stb#1578] [Q&A] [answered]
Hi, I recently begun researching about how to load gifs with stb_image. I came across this solution: https://github.com/opensoldat/opensoldat/blob/develop/client/libs/stb/src/stb.c but I am unsure how to use and implement it into my c++ project, I would really appreciate some guidance and help.

**Comment by paulwratt** [comment, marked as answer]
look in this project, its also used with zlib:
https://github.com/jart/cosmopolitan/blob/master/third_party/stb/stb_image.c

this is a tutorial on how to load images:
https://solarianprogrammer.com/2019/06/10/c-programming-reading-writing-images-stb_image-libraries/

I found this, TinyObjectLoader Viewer example:
https://android.googlesource.com/platform/external/tinyobjloader/+/HEAD/examples/viewer/stb_image.h

>       Animated GIF still needs a proper API, but here's one way to do it:
>          http://gist.github.com/urraka/685d9a6340b26b830d49

**Reply by Platforming-Mayhem** [reply]
The only issue is that I have no idea how to access the other frames of the animated gif.
